uint16 vehicleid
int8 len
int8[]  data